src/compat/dbtype_function.i

```diff
@@ -0,0 +1,2054 @@
+  CHECK_1ARG_NULL (value);
+#endif
+  if (value->domain.general_info.is_null || value->domain.general_info.type == DB_TYPE_ERROR)
+    {
+      return NULL;
+    }
+
+  switch (value->data.ch.info.style)
+    {
+    case SMALL_STRING:
+      str = (char *) value->data.ch.sm.buf;
+      break;
+    case MEDIUM_STRING:
+      str = value->data.ch.medium.buf;
+      break;
+    case LARGE_STRING:
+      /* Currently not implemented */
+      str = NULL;
+      break;
+    }
+
+  return str;
+}
+
+/*
+ * db_get_float() -
+ * return :
+ * value(in):
+ */
+DB_MACRO_INLINE float
+db_get_float (const DB_VALUE * value)
+{
+#if defined(API_ACTIVE_CHECKS)
+  CHECK_1ARG_ZERO (value);
+#endif
+  assert (value->domain.general_info.type == DB_TYPE_FLOAT);
+
+  return value->data.f;
+}
+
+/*
+ * db_get_double() -
+ * return :
+ * value(in):
+ */
+DB_MACRO_INLINE double
+db_get_double (const DB_VALUE * value)
+{
+#if defined(API_ACTIVE_CHECKS)
+  CHECK_1ARG_ZERO (value);
+#endif
+  assert (value->domain.general_info.type == DB_TYPE_DOUBLE);
+
+  return value->data.d;
+}
+
+/*
+ * db_get_object() -
+ * return :
+ * value(in):
+ */
+DB_MACRO_INLINE DB_OBJECT *
+db_get_object (const DB_VALUE * value)
+{
+#if defined(API_ACTIVE_CHECKS)
+  CHECK_1ARG_NULL (value);
+#endif
+  if (value->domain.general_info.is_null || value->domain.general_info.type == DB_TYPE_ERROR)
+    {
+      return NULL;
+    }
+  else
+    {
+      return value->data.op;
+    }
+}
+
+/*
+ * db_get_set() -
+ * return :
+ * value(in):
+ */
+DB_MACRO_INLINE DB_SET *
+db_get_set (const DB_VALUE * value)
+{
+#if defined(API_ACTIVE_CHECKS)
+  CHECK_1ARG_NULL (value);
+#endif
+  if (value->domain.general_info.is_null || value->domain.general_info.type == DB_TYPE_ERROR)
+    {
+      return NULL;
+    }
+  else
+    {
+      return value->data.set;
+    }
+}
+
+/*
+ * db_get_midxkey() -
+ * return :
+ * value(in):
+ */
+DB_MACRO_INLINE DB_MIDXKEY *
+db_get_midxkey (const DB_VALUE * value)
+{
+#if defined(API_ACTIVE_CHECKS)
+  CHECK_1ARG_NULL (value);
+#endif
+  if (value->domain.general_info.is_null || value->domain.general_info.type == DB_TYPE_ERROR)
+    {
+      return NULL;
+    }
+  else
+    {
+      return (DB_MIDXKEY *) (&(value->data.midxkey));
+    }
+}
+
+/*
+ * db_get_pointer() -
+ * return :
+ * value(in):
+ */
+DB_MACRO_INLINE void *
+db_get_pointer (const DB_VALUE * value)
+{
+#if defined(API_ACTIVE_CHECKS)
+  CHECK_1ARG_NULL (value);
+#endif
+  if (value->domain.general_info.is_null || value->domain.general_info.type == DB_TYPE_ERROR)
+    {
+      return NULL;
+    }
+  else
+    {
+      return value->data.p;
+    }
+}
+
+/*
+ * db_get_time() -
+ * return :
+ * value(in):
+ */
+DB_MACRO_INLINE DB_TIME *
+db_get_time (const DB_VALUE * value)
+{
+#if defined(API_ACTIVE_CHECKS)
+  CHECK_1ARG_NULL (value);
+#endif
+  assert (value->domain.general_info.type == DB_TYPE_TIME || value->domain.general_info.type == DB_TYPE_TIMELTZ);
+
+  return ((DB_TIME *) (&value->data.time));
```

Comment:
Remove cast. Same for:

  - db_get_timetz
  - db_get_timestamp
  - db_get_timestamptz
  - db_get_datetime
  - db_get_datetimetz
  - db_get_date
  - db_get_monetary

Reply:
We have some issues with removing these casts. Here, the casts are used for const stripping. Therefore we need to decide how will we handle them since, by plain removing the casts, we cannot modify the returned values because of the const parameters. I added a `todo` which needs to be reviewed in the future.